From e971a94774239bcdee45eddaea0ddb89149694d7 Mon Sep 17 00:00:00 2001
From: Eric Yang <eric.yang2@amd.com>
Date: Wed, 1 Jun 2016 14:36:35 -0400
Subject: [PATCH 0517/1722] drm/amd/dal: fix pipe split policy for multiple
 stream case

The secondary_pipe member was not set to NULL pre-split

Also cleaned up logging and ifdef

Signed-off-by: Eric Yang <eric.yang2@amd.com>
Acked-by: Jordan Lazare <Jordan.Lazare@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
Signed-off-by: Kalyan Alle <kalyan.alle@amd.com>
---
 drivers/gpu/drm/amd/dal/dc/core/dc.c | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/amd/dal/dc/core/dc.c b/drivers/gpu/drm/amd/dal/dc/core/dc.c
index e7f37bc..a7b4800 100644
--- a/drivers/gpu/drm/amd/dal/dc/core/dc.c
+++ b/drivers/gpu/drm/amd/dal/dc/core/dc.c
@@ -875,9 +875,8 @@ bool dc_commit_surfaces_to_target(
 	struct core_target *target = DC_TARGET_TO_CORE(dc_target);
 	struct dc_target_status *target_status = NULL;
 	struct validate_context *context;
-#ifdef DIAGS_BUILD
 	struct validate_context *temp_context;
-#endif
+
 	int current_enabled_surface_count = 0;
 	int new_enabled_surface_count = 0;
 	bool is_mpo_turning_on = false;
@@ -955,7 +954,6 @@ bool dc_commit_surfaces_to_target(
 		goto unexpected_fail;
 	}
 
-#ifdef DIAGS_BUILD
         if (core_dc->res_pool->funcs->apply_clk_constraints) {
                 temp_context = core_dc->res_pool->funcs->apply_clk_constraints(
                                 core_dc,
@@ -969,7 +967,6 @@ bool dc_commit_surfaces_to_target(
                 dm_free(context);
                 context = temp_context;
         }
-#endif
 
 	if (prev_disp_clk < context->bw_results.dispclk_khz ||
 		(is_mpo_turning_on &&
-- 
2.7.4

